### PR TITLE
Advanced Search Stimulus.js

### PR DESCRIPTION
**Pull request by martinemde**
This was another quick conversion. I noticed also that bad input on the downloads and updated fields would produce query errors, so a made a simple validation checker that helps you input a valid value. The validations are not very strict to avoid causing problems, but it really helps make less mistakes on the form.

### CODE REVIEW

File: app/views/searches/advanced.html.erb [hunk 1]
```diff
@@ -1,8 +1,16 @@
 <% @title = t("advanced_search") %>
 
-<div class="home__search-wrap" role="search">
-  <%= form_tag search_url, id: "advanced-search", method: :get do %>
-    <%= search_field_tag :query, params[:query], placeholder: t("layouts.application.header.search_gem_html"), autofocus: current_page?(root_url), id: "query", class: "home__search" %>
+<div class="home__search-wrap" role="search" data-controller="search">
+  <%= form_tag search_url, id: "advanced-search", method: :get, data: { search_target: "form" } do %>
```
Comment (simi) [inline app/views/searches/advanced.html.erb:4]: Is this data attribute still used? 🤔

Comment (martinemde) [reply app/views/searches/advanced.html.erb:4]: oh good catch, I'll remove it. I accessed the form through queryTarget.form instead.

File: app/views/searches/advanced.html.erb [hunk 2]
```diff
@@ -11,18 +19,18 @@
 
   <dl class="search-fields">
     <dt><%= label_tag :name, t(".name"), class: "form__label" %></dt>
-    <dd><%= text_field_tag :name, "", placeholder: "active OR action", class: "form__input"%></dd>
+    <dd><%= text_field_tag :name, "", placeholder: "active OR action", class: "form__input", data: { search_target: "attribute", action: "input->search#input keydown.enter->search#submit" } %></dd>
```
Comment (simi) [inline app/views/searches/advanced.html.erb:22]: I'm total Stimulus.js noob, probably I'm missing something, but wouldn't be 1 event of "change" (instead input and keydown.enter) enough?

Comment (simi) [reply app/views/searches/advanced.html.erb:22]: sorry, I mean "input" only

Comment (simi) [reply app/views/searches/advanced.html.erb:22]: ahh, I do follow now, looks ok

**Review by simi — APPROVED** [review]

**Comment by simi** [comment]
💪

**Pull request merged**